lib/ansible/modules/remote_management/cpm/cpm_serial_port_config.py

```diff
@@ -0,0 +1,345 @@
+        portname=dict(type='str', required=False, default=None),
+        baud=dict(type='int', required=False, default=None, choices=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+        handshake=dict(type='int', required=False, default=None, choices=[0, 1, 2, 3]),
+        stopbits=dict(type='int', required=False, default=None, choices=[0, 1]),
+        parity=dict(type='int', required=False, default=None, choices=[0, 1, 2, 3, 4, 5]),
+        mode=dict(type='int', required=False, default=None, choices=[0, 1, 2, 3, 4]),
+        cmd=dict(type='int', required=False, default=None, choices=[0, 1]),
+        seq=dict(type='int', required=False, default=None, choices=[1, 2, 3]),
+        tout=dict(type='int', required=False, default=None, choices=[0, 1, 2, 3, 4, 5]),
+        echo=dict(type='int', required=False, default=None, choices=[0, 1]),
+        break_allow=dict(type='int', required=False, default=None, choices=[0, 1]),
+        logoff=dict(type='str', required=False, default=None),
+        use_https=dict(type='bool', default=True),
+        validate_certs=dict(type='bool', default=True),
+        use_proxy=dict(type='bool', default=False)
+    )
+
+    result = dict(
+        changed=False,
+        data=''
+    )
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    auth = to_text(base64.b64encode(to_bytes('{0}:{1}'.format(to_native(module.params['cpm_username']), to_native(module.params['cpm_password'])),
+                   errors='surrogate_or_strict')))
+
+    if module.params['use_https'] is True:
+        protocol = "https://"
+    else:
+        protocol = "http://"
+
+    fullurl = ("%s%s/api/v2/config/serialports?ports=%s" % (protocol, to_native(module.params['cpm_url']), to_native(module.params['port'])))
+    method = 'GET'
+    read_data = ""
+    try:
+        response = open_url(fullurl, data=None, method=method, validate_certs=module.params['validate_certs'], use_proxy=module.params['use_proxy'],
+                            headers={'Content-Type': 'application/json', 'Authorization': "Basic %s" % auth})
+
+    except HTTPError as e:
+        fail_json = dict(msg='GET: Received HTTP error for {0} : {1}'.format(fullurl, to_native(e)), changed=False)
+        module.fail_json(**fail_json)
+    except URLError as e:
+        fail_json = dict(msg='GET: Failed lookup url for {0} : {1}'.format(fullurl, to_native(e)), changed=False)
+        module.fail_json(**fail_json)
+    except SSLValidationError as e:
+        fail_json = dict(msg='GET: Error validating the server''s certificate for {0} : {1}'.format(fullurl, to_native(e)), changed=False)
+        module.fail_json(**fail_json)
+    except ConnectionError as e:
+        fail_json = dict(msg='GET: Error connecting to {0} : {1}'.format(fullurl, to_native(e)), changed=False)
+        module.fail_json(**fail_json)
+
+    read_data = json.loads(response.read())
+    payload = assemble_json(module, read_data)
+
```

Comment:
Minor change to the check mode logic here since it's not properly reporting a change if a change would be made when run in check mode, even though we have that information available.

This looks like a bigger change than it is because I nested it one level.
```suggestion
    result['data'] = read_data

    if module.check_mode:
        if payload is not None:
            result['changed'] = True
    else:
        if payload is not None:
            fullurl = ("%s%s/api/v2/config/serialports" % (protocol, to_native(module.params['cpm_url'])))
            method = 'POST'

            try:
                response = open_url(fullurl, data=payload, method=method, validate_certs=module.params['validate_certs'], use_proxy=module.params['use_proxy'],
                                    headers={'Content-Type': 'application/json', 'Authorization': "Basic %s" % auth})

            except HTTPError as e:
                fail_json = dict(msg='POST: Received HTTP error for {0} : {1}'.format(fullurl, to_native(e)), changed=False)
                module.fail_json(**fail_json)
            except URLError as e:
                fail_json = dict(msg='POST: Failed lookup url for {0} : {1}'.format(fullurl, to_native(e)), changed=False)
                module.fail_json(**fail_json)
            except SSLValidationError as e:
                fail_json = dict(msg='POST: Error validating the server''s certificate for {0} : {1}'.format(fullurl, to_native(e)), changed=False)
                module.fail_json(**fail_json)
            except ConnectionError as e:
                fail_json = dict(msg='POST: Error connecting to {0} : {1}'.format(fullurl, to_native(e)), changed=False)
                module.fail_json(**fail_json)

            result['changed'] = True
            result['data'] = json.loads(response.read())
```